test/ApplicationTest.php

```diff
@@ -128,6 +129,18 @@ public function testPipeCanAcceptAPathArgument($middleware)
         $this->assertNull($this->app->pipe('/foo', $middleware));
     }
 
+    public function testPipeNonSlashPathOnNonStringPipeProduceTypeError()
+    {
+        $this->expectException(TypeError::class);
```

Comment:
Please move this just before line which should raise the exception. Otherwise it's not that clear. I can see that the exception can't be throw in other place, but it's good convention to have it just before the place where it should be thrown.

Reply:
updated.